src/Common/src/Interop/Windows/SChannel/Interop.Sec_Application_Protocols.cs

```diff
@@ -38,27 +36,23 @@ public static unsafe byte[] ToByteArray(List<SslApplicationProtocol> application
             }
 
             Sec_Application_Protocols protocols = new Sec_Application_Protocols();
-            protocols.ProtocolListsSize = (uint)(ProtocolListConstSize + protocolListSize);
-            protocols.ProtocolExtenstionType = ApplicationProtocolNegotiationExt.ALPN;
+            protocols.ProtocolListsSize = (uint)((sizeof(Sec_Application_Protocols) - sizeof(uint)) + protocolListSize);
+            protocols.ProtocolExtensionType = ApplicationProtocolNegotiationExt.ALPN;
             protocols.ProtocolListSize = (short)protocolListSize;
 
-            Span<byte> pBuffer = new byte[protocolListSize];
+            byte[] buffer = new byte[sizeof(Sec_Application_Protocols) + protocolListSize];
```

Comment:
This appears to be a larger value than it was before.  Was it broken before?

Reply:
Oh, I see, this is replacing a second allocation made below. Nice.

Reply:
The original code allocated two buffers: First temporary buffer was `protocolListSize`, and second final buffer was `sizeof(Sec_Application_Protocols) + protocolListSize`. I got rid of the temporary buffer allocation while I was on it.